### ISSUE TITLE
Support for macros, even a simplified variant, would be beneficial

### ISSUE THREAD

**Issue by bengalack** [theNestruo/z80-asm-meter-vscode#73]
I have some macros which I use with SDCC (sdasz80). It would be nice if they could be recognized as "custom entries".

Here are a few examples of macros that may appear anywhere in an asm-file.

```
; ------------------------------
;
.macro rightShift3_HL_trashA
	LD 	A, L
	SRL	H
	RRA
	SRL	H
	RRA
	SRL 	H
	RRA
	LD	L, A
.endm

; ------------------------------
;
.macro ld____ixh_d
	.db 0xDD, 0x62 	; 10 MSX cycles
.endm

; ------------------------------
; Macro that is used only a few places
; Uses another macro (not included here) and is using repeat too
;
.macro writeSpriteAttrTableRamToVramMasksOnly
	writeSpriteAttrTableRamToVramMasksPrepp

.rept 7
	writeSpriteAttrTableRamToVramMasksOne
.endm
	outi

.endm

; ------------------------------
; This one is different in that it has a conditional
;
.macro addAtoHL ?noparameter
	ADD 	A, L
	LD 	L, A
	JR 	NC, noparameter
	INC 	H
noparameter:
.endm

```

and using/inserting these in your code, is just by naming them. So, the above may be used like this:

```
	...
	addAtoHL
	rightShift3_HL_trashA
	call 	somefunction
	
	ld	d, l
	ld____ixh_d
	...
```

If this cannot be parsed directly, it would be beneficial to be able to manually maintain a custom-section in a settings-file (hopefully no need to restart VSCode each time there is a change).

As I do have some macros that are much longer too, and macros that nest other macros (like `writeSpriteAttrTableRamToVramMasksOnly`), the sjasmplus-extension may be seem a bit complex to use? Maybe we can "just extend the instruction set", like this:

```
rightShift3_HL_trashA 55, 11 bytes
ld____ixh_d 10, 2 bytes
addAtoHL 28/23, 5 bytes
```

**Comment by bengalack** [comment]
I think I only need the last part of (4). The timing is more accurate (as you point out), and the size/bytes may be as well. I never/seldom use the byte-count anyway, so atm, I have no requirements there. 

Adding these numbers manually is generally very easy as you just use this tool on the section of the code and get the values from that.

Also, sometimes I have macros called `repeat32outis` and even `repeat256outis`, and adding these to the sjasmplus-format doesn't seem right :-O (yes, I use megarom, and space is not the biggest concern)

**Comment by theNestruo** [comment]
Ok!
Next week I'll try to do (1), (2) and (4) to cover most use cases but giving preference to (4), and I hope to have it finished by the weekend.
Have a nice week!

**Comment by theNestruo** [comment]
I have a preliminary implementation of the support for user-defined macros. I still need to improve edge case scenarios (macros without any optional field in their definition, incoherences between _size_ and _instructions_ fields, etc.), but the base code is ready.

I have done a couple of tests with these definitions:
```jsonc
	"z80-asm-meter.macros": [
		{
			// Macro defined with instructions
			"name": "rightShift3_HL_trashA",
			"instructions": [ "LD A, L", "SRL H", "RRA", "SRL H", "RRA", "SRL H", "RRA", "LD L, A" ]
		},
		{
			// User-provided timing and size (no instructions)
			"name": "ld____ixh_d",
			"msx": 10,
			"size": 2
		},
		{
			// User-provided timing overriding instructions-based computed timing
			"name": "addAtoHL",
			"msx": 23,
			"instructions": [ "ADD A, L", "LD L, A", "JR NC, noparameter", "INC H" ]
		}
	],
```

Please download this nightly build (https://github.com/theNestruo/z80-asm-meter-vscode/suites/5294656611/artifacts/164176675) and tell me if everything is looking good or if any problem arises.

**Comment by bengalack** [comment]
Awesome, this works really smooth.

First I got this: "Unable to install extension 'thenestruo.z80-asm-meter' as it is not compatible with VS Code '1.63.2'."

So I just updated, and things work really well. But why would I want to type in the instructions? Seems like more work than just add the "msx" and "size" :)

Anyways: Good work

**Comment by bengalack** [comment]
> It would also support `[ "ld____ixh_d", "LD IXH, D" ]` (will be detected as the instruction `LD IXh,p`), but `[ "ld____ixh_d", "db 0xDD", "db 0x62" ]` won't be properly parsed. 

A little late, but just to comment on this one as well: This is fine by me, you already have support for undocumented z80 instructions (unlike SDCC), so that solves it.

**Comment by theNestruo** [comment]
> First I got this: "Unable to install extension 'thenestruo.z80-asm-meter' as it is not compatible with VS Code '1.63.2'."
>
> So I just updated, and things work really well. 
>
Yup. Toolchain screams when something is not on the latest version. I don't think I'm using anything that didn't exist like 10 or 20 versions ago but... ¯\(°_o)/¯
 
> But why would I want to type in the instructions? Seems like more work than just add the "msx" and "size" :)
> 
Most likely if you are tweaking your macro and don't want to be updating the timing every time. Or if you prefer to have the instructions to quickly check against the macro in the code if it's the same. Don't really know the exact reason, but as long as it is optional, having the possibility is better than not having it :)

> Awesome, this works really smooth.
>
> Anyways: Good work
>
Thank you very much!
I'm happy to know my extension is useful! Yay!!

**Comment by theNestruo** [comment]
Macro support included in version 3.0.0
